### PR TITLE
UGENE-6995. Go to read start MCA editor action

### PR DESCRIPTION
**Pull request by yalgaer**

### CODE REVIEW

File: src/corelibs/U2View/src/ov_msa/McaEditor.cpp [hunk 1]
```diff
@@ -234,9 +226,20 @@ void McaEditor::initActions() {
     showOverviewAction->setChecked(overviewVisible);
     ui->getOverviewArea()->setVisible(overviewVisible);
     changeFontAction->setText(tr("Change characters font..."));
+
+    gotoSelectedReadAction = new QAction(tr("Goto selected read"), this);
```
Comment (atiunov) [inline src/corelibs/U2View/src/ov_msa/McaEditor.cpp]: "Go to"?

File: src/corelibs/U2View/src/ov_msa/McaEditor.cpp [hunk 2]
```diff
@@ -234,9 +226,20 @@ void McaEditor::initActions() {
...
+    gotoSelectedReadAction->setObjectName("centerReadStartAction");
+    connect(gotoSelectedReadAction, SIGNAL(triggered()), SLOT(sl_gotoSelectedRead()));
```
Comment (atiunov) [inline src/corelibs/U2View/src/ov_msa/McaEditor.cpp]: Set the action initial state. Just to be sure.

File: src/corelibs/U2View/src/ov_msa/McaEditor.cpp [hunk 3]
```diff
@@ -333,4 +338,14 @@ void McaEditor::addEditMenu(QMenu *menu) {
     editMenu->addAction(ui->getRedoAction());
 }
 
+void McaEditor::sl_gotoSelectedRead() {
+    MaEditorSelection selection = getSelection();
+    int rowIndex = selection.y();
+    CHECK(selection.height() > 0 && rowIndex >= 0 && rowIndex < maObject->getNumRows(), );
+
+    MultipleChromatogramAlignmentRow mcaRow = getMaObject()->getMcaRow(rowIndex);
+    qint64 rowStartPos = mcaRow->isComplemented() ? mcaRow->getCoreEnd() : mcaRow->getCoreStart();
```
Comment (atiunov) [inline src/corelibs/U2View/src/ov_msa/McaEditor.cpp]: Both "getCoreEnd" and "getCoreStart" return "int". But some functions that are called inside these methods return qint64. Neither MCA Editor, nor MSA Editor could correctly process very long rows, so I offer to use "int" for the base positions.

**Review by atiunov — CHANGES_REQUESTED** [review]

File: src/corelibs/U2View/src/ov_msa/McaEditor.cpp (hunk 1 above)
Comment (yalgaer) [reply src/corelibs/U2View/src/ov_msa/McaEditor.cpp]: Fixed

File: src/corelibs/U2View/src/ov_msa/McaEditor.h [hunk 4]
```diff
@@ -85,14 +83,24 @@ private slots:
...
+    void updateActions() override;
 
     QAction *showChromatogramsAction;
     QAction *showGeneralTabAction;
     QAction *showConsensusTabAction;
 
+    /**
+     * When activated MCA editor moves start of the currently selected read into the view.
+     * If the selection contains multiple reads - selects the first one.
+     */
```
Comment (atiunov) [inline src/corelibs/U2View/src/ov_msa/McaEditor.h]: Please, add information about complementary reads.

**Review by atiunov — CHANGES_REQUESTED** [review]

**Review by atiunov — APPROVED** [review]

**Pull request merged**